test/should.js

```diff
@@ -1407,6 +1407,16 @@ describe('should', function() {
 
     ({ 'foo.bar[]': 'baz'}).should.have.nested.property('foo\\.bar\\[\\]');
 
+    ({a:1}).should.have.nested.property('a');
+
+    (function () {
+      ({a:1}).should.have.nested.property('{a:1}');
+    }).should.not.throw('the argument to property must be a string when using nested syntax');
+
+    (function () {
+      ({a:1}).should.have.nested.property({'a':'1'});
+    }).should.throw('the argument to property must be a string when using nested syntax');
+
```

Comment:
One more test should be added here to the `should` interface tests in order to verify the error message "the argument to property must be a string, number, or symbol". This test can closely resemble the `expect({a:1}).to.have.property(null);` test that you added on the `expect` interface, except using the `should` interface this time instead of `expect`. (And using the `err` helper function).